webapp/channels/src/components/user_settings/modal/user_settings_modal.tsx

```diff
@@ -308,45 +350,74 @@ class UserSettingsModal extends React.PureComponent<Props, State> {
                     >
                         {modalTitle}
                     </Modal.Title>
-                </Modal.Header>
-                <Modal.Body ref={this.modalBodyRef}>
-                    <div className='settings-table'>
-                        <div className='settings-links'>
-                            <SettingsSidebar
-                                tabs={this.props.isContentProductSettings ? this.getUserSettingsTabs() : this.getProfileSettingsTab()}
-                                pluginTabs={this.props.isContentProductSettings ? this.getPluginsSettingsTab() : []}
-                                activeTab={this.state.active_tab}
-                                updateTab={this.updateTab}
+
+                    {
+                        this.props.adminMode &&
+                        <div className='adminModeBadge'>
+                            <FormattedMessage
+                                id='userSettings.adminMode.admin_mode_badge'
+                                defaultMessage='Admin Mode'
                             />
                         </div>
-                        <div className='settings-content minimize-settings'>
-                            <UserSettings
-                                activeTab={this.state.active_tab}
-                                activeSection={this.state.active_section}
-                                updateSection={this.updateSection}
-                                updateTab={this.updateTab}
-                                closeModal={this.closeModal}
-                                collapseModal={this.collapseModal}
-                                setEnforceFocus={(enforceFocus?: boolean) => this.setState({enforceFocus})}
-                                setRequireConfirm={
-                                    (requireConfirm?: boolean, customConfirmAction?: () => () => void) => {
-                                        this.requireConfirm = requireConfirm!;
-                                        this.customConfirmAction = customConfirmAction!;
+                    }
+                </Modal.Header>
+                <Modal.Body ref={this.modalBodyRef}>
+                    {
+                        this.props.adminMode &&
+                        <SmartLoader
+                            loading={this.props.adminMode && (!this.props.userPreferences || !this.props.currentUser)}
```

Comment:
This is definitely a case where having the other user be `this.props.currentUser` confused me a bit.

Also, checking `this.props.adminMode` here is a bit redundant here because it's used outside of the component, but 0/5 on actually changing that since it keeps it clear how this is meant to be used